### PR TITLE
🎉 New Source: OpsGenie

### LINKED ISSUE #14566: New Source: OpsGenie
## Tell us about the new connector you’d like to have

* Which source and which destination?

We want to source information from OpsGenie into our Snowflake warehouse.

* Do you need a specific version of the underlying data source e.g: you specifically need support for an older version of the API or DB?

No. The current cloud platform API version is adequate.

## Describe the context around this new connector

* Why do you need this integration? How does your team intend to use the data? This helps us understand the use case.

We ingest a lot of operational data into our data platform using other Airbyte connectors. We want to add OpsGenie data using our existing tools and services to keep our operational life simple.

* How often do you want to run syncs?

Daily

* If this is an API source connector, which entities/endpoints do you need supported?

Priority is Team, User, Alerts, and Incidents.

APIs listed on the [OpsGenie Documentation](https://docs.opsgenie.com/docs/api-overview)

* If the connector is for a paid service, can we name you as a mutual user when we subscribe for an account? Which company should we name? 

## Describe the alternative you are considering or using

We would need to build custom extract job/s and coordinate with Airflow.

## Are you willing to submit a PR?

Yes

### PR DESCRIPTION
**Pull request by adamschmidt**
## What

This change adds an initial release of a source connector for the OpsGenie cloud monitoring platform.

## How

OpsGenie has a rich set of APIs to read information from the platform. The initial release of the connector extracts information about the following:

* Incidents
* Alerts
* Teams
* Team Members & their teams
* Services
* Integrations

Closes #14566

## Recommended reading order
1. `airbyte-integrations/connectors/source-opsgenie/source_opsgenie/source.py`
2. `airbyte-integrations/connectors/source-opsgenie/source_opsgenie/streams.py`

## 🚨 User Impact 🚨

No breaking changes

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

```
Test session starts (platform: darwin, Python 3.9.13, pytest 6.2.5, pytest-sugar 0.9.5)
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/adam.schmidt/Development/github/airbyte/airbyte/airbyte-integrations/connectors/source-opsgenie/.hypothesis/examples')
rootdir: /Users/adam.schmidt/Development/github/airbyte/airbyte, configfile: pytest.ini
plugins: sugar-0.9.5, requests-mock-1.9.3, mock-3.6.1, hypothesis-6.54.4, timeout-1.4.2, cov-3.0.0
collecting ... 
 airbyte-integrations/connectors/source-opsgenie/unit_tests/test_source.py::SourceOpsgenieTest.test_check_connection ✓                                                                                                    10% █         
 airbyte-integrations/connectors/source-opsgenie/unit_tests/test_source.py::SourceOpsgenieTest.test_stream_count ✓                                                                                                        20% ██        
 airbyte-integrations/connectors/source-opsgenie/unit_tests/test_stream.py::TeamsStreamTestCase.test_teams_list ✓                                                                                                         30% ███       
 airbyte-integrations/connectors/source-opsgenie/unit_tests/test_stream.py::UsersStreamTestCase.test_users_list ✓                                                                                                         40% ████      
 airbyte-integrations/connectors/source-opsgenie/unit_tests/test_stream.py::ServicesStreamTestCase.test_services_list ✓                                                                                                   50% █████     
 airbyte-integrations/connectors/source-opsgenie/unit_tests/test_stream.py::IntegrationsStreamTestCase.test_services_list ✓                                                                                               60% ██████    
 airbyte-integrations/connectors/source-opsgenie/unit_tests/test_stream.py::UserTeamsStreamTestCase.test_user_teams_list ✓                                                                                                70% ███████   
 airbyte-integrations/connectors/source-opsgenie/unit_tests/test_stream.py::AlertsStreamTestCase.test_alerts_list ✓                                                                                                       80% ████████  
 airbyte-integrations/connectors/source-opsgenie/unit_tests/test_stream.py::IncidentsStreamTestCase.test_alerts_list ✓                                                                                                    90% █████████ 
 airbyte-integrations/connectors/source-opsgenie/unit_tests/unit_test.py::AuthenticatorTestCase.test_token ✓                                                                                                             100% ██████████
=========================================================================================================== warnings summary ===========================================================================================================
airbyte-integrations/connectors/source-opsgenie/unit_tests/test_source.py: 7 warnings
airbyte-integrations/connectors/source-opsgenie/unit_tests/test_stream.py: 8 warnings
  /Users/adam.schmidt/Development/github/airbyte/airbyte/airbyte-integrations/connectors/source-opsgenie/.venv/lib/python3.9/site-packages/airbyte_cdk/sources/streams/http/http.py:41: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    self._authenticator: HttpAuthenticator = NoAuth()

airbyte-integrations/connectors/source-opsgenie/unit_tests/test_source.py: 7 warnings
airbyte-integrations/connectors/source-opsgenie/unit_tests/test_stream.py: 8 warnings
  /Users/adam.schmidt/Development/github/airbyte/airbyte/airbyte-integrations/connectors/source-opsgenie/.venv/lib/python3.9/site-packages/deprecated/classic.py:173: DeprecationWarning: Call to deprecated class HttpAuthenticator. (Use requests.auth.AuthBase instead) -- Deprecated since version 0.1.20.
    return old_new1(cls, *args, **kwargs)

-- Docs: https://docs.pytest.org/en/stable/warnings.html

Results (0.59s):
      10 passed

```

</details>

<details><summary><strong>Acceptance</strong></summary>

```
[+] Building 2.7s (17/17) FINISHED                                                                                                                                                                                                                               
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                        0.0s
 => => transferring dockerfile: 92B                                                                                                                                                                                                                         0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                           0.0s
 => => transferring context: 34B                                                                                                                                                                                                                            0.0s
 => [internal] load metadata for docker.io/library/python:3.9.11-alpine3.15                                                                                                                                                                                 2.6s
 => [auth] library/python:pull token for registry-1.docker.io                                                                                                                                                                                               0.0s
 => [internal] load build context                                                                                                                                                                                                                           0.0s
 => => transferring context: 1.52kB                                                                                                                                                                                                                         0.0s
 => [base 1/1] FROM docker.io/library/python:3.9.11-alpine3.15@sha256:45ddd216e6b4efee0617e15d541e9148ffd6898203fcbe86a9f5bf906ce7837f                                                                                                                      0.0s
 => CACHED [builder 1/4] WORKDIR /airbyte/integration_code                                                                                                                                                                                                  0.0s
 => CACHED [builder 2/4] RUN apk --no-cache upgrade     && pip install --upgrade pip     && apk --no-cache add tzdata build-base                                                                                                                            0.0s
 => CACHED [builder 3/4] COPY setup.py ./                                                                                                                                                                                                                   0.0s
 => CACHED [builder 4/4] RUN pip install --prefix=/install .                                                                                                                                                                                                0.0s
 => CACHED [stage-2 2/7] COPY --from=builder /install /usr/local                                                                                                                                                                                            0.0s
 => CACHED [stage-2 3/7] COPY --from=builder /usr/share/zoneinfo/Etc/UTC /etc/localtime                                                                                                                                                                     0.0s
 => CACHED [stage-2 4/7] RUN echo "Etc/UTC" > /etc/timezone                                                                                                                                                                                                 0.0s
 => CACHED [stage-2 5/7] RUN apk --no-cache add bash                                                                                                                                                                                                        0.0s
 => CACHED [stage-2 6/7] COPY main.py ./                                                                                                                                                                                                                    0.0s
 => CACHED [stage-2 7/7] COPY source_opsgenie ./source_opsgenie                                                                                                                                                                                             0.0s
 => exporting to image                                                                                                                                                                                                                                      0.0s
 => => exporting layers                                                                                                                                                                                                                                     0.0s
 => => writing image sha256:9b81438c15a7964f87705597eabca64273f474840544123be16a68d5f69ffd53                                                                                                                                                                0.0s
 => => naming to docker.io/airbyte/source-opsgenie:dev                                                                                                                                                                                                      0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
latest: Pulling from airbyte/source-acceptance-test
Digest: sha256:58baf665484234a569cd0059e4c4d9de19ced79251023639989aaa4b18794185
Status: Image is up to date for airbyte/source-acceptance-test:latest
docker.io/airbyte/source-acceptance-test:latest
Test session starts (platform: linux, Python 3.9.11, pytest 6.2.5, pytest-sugar 0.9.5)
rootdir: /test_input
plugins: timeout-1.4.2, sugar-0.9.5, mock-3.6.1, hypothesis-6.54.4, requests-mock-1.9.3, cov-3.0.0
collecting ... 
 test_core.py ✓✓✓✓✓✓✓✓✓✓s✓✓✓✓✓✓✓✓✓✓✓s✓✓                                                                                                                                                                                                            86% ████████▋ 
 test_full_refresh.py ✓                                                                                                                                                                                                                            90% █████████ 
 test_incremental.py ✓✓✓                                                                                                                                                                                                                          100% ██████████

==================================================================================================================== short test summary info ====================================================================================================================
SKIPPED [1] source_acceptance_test/tests/test_core.py:51: The previous connector image could not be retrieved.
SKIPPED [1] source_acceptance_test/tests/test_core.py:229: The previous connector image could not be retrieved.

Results (1144.95s):
      27 passed
       2 skipped
```

</details>

### CODE REVIEW

**Review by vincentkoc — COMMENTED** [review]
@adamschmidt please bare with me as I’ll be taking a look at this PR as part of the maintainers program.

File: airbyte-integrations/connectors/source-opsgenie/integration_tests/acceptance.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import pytest
+
+pytest_plugins = ("source_acceptance_test.plugin",)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def connector_setup():
+    """This fixture is a placeholder for external resources that acceptance test might require."""
+    # TODO: setup test dependencies if needed. otherwise remove the TODO comments
```
Comment (koconder) [inline airbyte-integrations/connectors/source-opsgenie/integration_tests/acceptance.py:14]: Drop all the `TODO`s in the code. Not sure if you have run the formatter on this code, can you confirm?

File: airbyte-integrations/connectors/source-opsgenie/source_opsgenie/source.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        authenticator = TokenAuthenticator(config["api_token"], auth_method="GenieKey")
+        return authenticator
+
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+
+        try:
+            auth = self.get_authenticator(config)
+            api_endpoint = f"https://{config['endpoint']}/v2/account"
+
+            response = requests.get(
```
Comment (koconder) [inline airbyte-integrations/connectors/source-opsgenie/source_opsgenie/source.py:32]: @adamschmidt have you thought about any retry logic or rate limiting? There are some examples in other connectors you can borrow logic from. It's not a must for the alpha but will eventually become necessary for reliability.

File: airbyte-integrations/connectors/source-opsgenie/source_opsgenie/util.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from source_opsgenie.streams import OpsgenieStream
+from airbyte_cdk.models import SyncMode
+
+def read_full_refresh(stream_instance: OpsgenieStream):
+    slices = stream_instance.stream_slices(sync_mode=SyncMode.full_refresh)
+    for _slice in slices:
+        records = stream_instance.read_records(stream_slice=_slice, sync_mode=SyncMode.full_refresh)
```
Comment (koconder) [inline airbyte-integrations/connectors/source-opsgenie/source_opsgenie/util.py:7]: Could you simplify with something like
```py
yield from stream_instance.read_records(stream_slice=_slice, sync_mode=SyncMode.full_refresh)
```

Comment (koconder) [inline airbyte-integrations/connectors/source-opsgenie/source_opsgenie/util.py:7]: Could you simplify with something like
```py
yield from stream_instance.read_records(stream_slice=_slice, sync_mode=SyncMode.full_refresh)
```

File: airbyte-integrations/connectors/source-opsgenie/integration_tests/catalog.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
+{
+  "streams": [
+    {
+      "name": "TODO fix this file",
```
Comment (koconder) [inline airbyte-integrations/connectors/source-opsgenie/integration_tests/catalog.json:4]: @adamschmidt what’s with this TODO?

File: airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/alerts.json [hunk 5]
```diff
@@ -0,0 +1,143 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": ["null", "string"]
+    },
+    "tinyId": {
```
Comment (koconder) [inline airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/alerts.json:8]: This is an `number` based on the API

File: airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/alerts.json [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    },
+    "description": {
+      "type": ["null", "string"]
+    },
+    "details": {
+      "type": ["null", "object"],
+      "properties": { },
+      "additionalProperties": true
+    },
+    "ownerTeamId": {
```
Comment (koconder) [inline airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/alerts.json:139]: @adamschmidt I can’t find `ownerTeamId` in the API docs, you have the right schemas identified? https://docs.opsgenie.com/docs/alert-api

File: airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/incidents.json [hunk 7]
```diff
@@ -0,0 +1,55 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": ["null", "string"]
+    },
+    "tinyId": {
```
Comment (koconder) [inline airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/incidents.json:8]: Use `number`

File: airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/services.json [hunk 8]
```diff
@@ -0,0 +1,38 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
```
Comment (koconder) [inline airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/services.json:4]: @adamschmidt again more fields located on the API https://docs.opsgenie.com/docs/incident-api

File: airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/integrations.json [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    "name": {
+      "type": ["null", "string"]
+    },
+    "enabled": {
+      "type": ["null", "boolean"]
+    },
+    "type": {
+      "type": ["null", "string"]
+    },
+    "teamId": {
```
Comment (koconder) [inline airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/integrations.json:17]: Team name is also supported in schema https://docs.opsgenie.com/docs/integration-api

File: airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/teams.json [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    "id": {
+      "type": ["null", "string"]
+    },
+    "name": {
+      "type": ["null", "string"]
+    },
+    "description": {
+      "type": ["null", "string"]
+    },
+    "links": {
```
Comment (koconder) [inline airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/teams.json:14]: Can’t find `links` but found members https://docs.opsgenie.com/docs/team-api

File: airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/users.json [hunk 11]
```diff
@@ -0,0 +1,62 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
```
Comment (koconder) [inline airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/users.json:2]: Missing fields such as `invitationDisabled` see https://docs.opsgenie.com/docs/user-api

File: airbyte-integrations/connectors/source-opsgenie/source_opsgenie/source.py (hunk 2 above)
Comment (koconder) [reply airbyte-integrations/connectors/source-opsgenie/source_opsgenie/source.py:32]: Considerations:
* Rate Limiting: https://docs.opsgenie.com/docs/api-rate-limiting
* Response Codes: https://docs.opsgenie.com/docs/response

File: docs/integrations/README.md [hunk 12]
```diff
@@ -110,6 +110,7 @@ For more information about the grading system, see [Product Release Stages](http
 | [Okta](sources/okta.md)                                                                     | Alpha                | Yes                 |
 | [OneSignal](sources/onesignal.md)                                                           | Alpha                | No                  |
 | [OpenWeather](sources/openweather.md)                                                       | Alpha                | No                  |
+| [OpsGenie](sources/opsgenie.md)                                                             | Alpha                | No                  |
```
Comment (koconder) [inline docs/integrations/README.md:113]: Rename as mentioned before

File: docs/integrations/sources/opsgenie.md [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+### Requirements
+
+* OpsGenie Account
+* OpsGenie API Key wih the necessary permissions \(described below\)
+
+### Setup guide
+
+Log into OpsGenie and then generate an [API Key](https://support.atlassian.com/opsgenie/docs/api-key-management/).
+
+Your token should have the `read` and `configuration access` access rights so that the connector can call the platform APIs it depends on for data.
```
Comment (koconder) [inline docs/integrations/sources/opsgenie.md:42]: Feel free to add link https://docs.opsgenie.com/docs/api-access-management

File: airbyte-integrations/connectors/source-opsgenie/source_opsgenie/source.py (hunk 2 above)
Comment (koconder) [reply airbyte-integrations/connectors/source-opsgenie/source_opsgenie/source.py:32]: This is important for the streams, not so much the initial connection checks

File: airbyte-integrations/connectors/source-opsgenie/source_opsgenie/source.py [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+                api_endpoint,
+                headers=auth.get_auth_header(),
+            )
+
+            return response.status_code == requests.codes.ok, None
+
+        except Exception as error:
+            return False, f"Unable to connect to OpsGenie API with the provided credentials - {repr(error)}"
```
[the remark this answers is not in the document]
Comment (koconder) [reply airbyte-integrations/connectors/source-opsgenie/source_opsgenie/source.py:40]: I.e. unable to connect due to rate limiting, bla bla.

File: airbyte-integrations/connectors/source-opsgenie/setup.py [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+    description="Source implementation for Opsgenie.",
+    author="Airbyte",
+    author_email="contact@airbyte.io",
+    packages=find_packages(),
+    install_requires=MAIN_REQUIREMENTS,
+    package_data={"": ["*.json", "*.yaml", "schemas/*.json", "schemas/shared/*.json"]},
+    extras_require={
+        "tests": TEST_REQUIREMENTS,
+    },
+)
```
Comment (koconder) [inline airbyte-integrations/connectors/source-opsgenie/setup.py:30]: Please reinstate:

```py


dependencies {
    implementation files(project(':airbyte-integrations:bases:source-acceptance-test').airbyteDocker.outputs)
}
```

**Review by vincentkoc — CHANGES_REQUESTED** [review]
@adamschmidt thanks for your contribution. Few things:

* Please provide integration tests, as I’m seeing failures on my side.
* See comments on various things, mostly on schema parity with Opsgenie docs and rename a few things

**Comment by adamschmidt** [comment]
Thanks for the feedback, I'll work my way through it today.

File: airbyte-integrations/connectors/source-opsgenie/setup.py (hunk 15 above)
Comment (adamschmidt) [reply airbyte-integrations/connectors/source-opsgenie/setup.py:30]: Is this a copypasta? I'm not sure where this Gradle snippet needs to go?

File: airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/alerts.json (hunk 6 above)
Comment (adamschmidt) [reply airbyte-integrations/connectors/source-opsgenie/source_opsgenie/schemas/alerts.json:139]: This was based on live data from the account I'm using for dev/testing. Perhaps it's an undocumented field?

I can confirm that it's being consistently populated in the data I'm retrieving in testing.

File: airbyte-integrations/connectors/source-opsgenie/setup.py (hunk 15 above)
Comment (koconder) [reply airbyte-integrations/connectors/source-opsgenie/setup.py:30]: @adamschmidt look at other connectors gradle files, i don’t think your gradle needs to be modified. Will go at the end after all other condition's.

**Comment by adamschmidt** [comment]
🤦 I synced the fork which closed this PR

**Comment by sajarin** [comment]
@adamschmidt can you open a new pr with your changes? We can continue the review there.

**Comment by koconder** [comment]
@adamschmidt when you open a new PR just tag me in this to review and I’ll get onto it ASAP :)

**Pull request closed without merging**